Fixtures/Miscellaneous/TestDiscovery/Async/Tests/AsyncTests/SwiftTests.swift

```diff
@@ -0,0 +1,17 @@
+import XCTest
+@testable import Async
+
+class AsyncTests: XCTestCase {
+
+    func testAsync() async {
+        XCTAssertFalse(Thread.isMainThread)
```

Comment:
Are we confident that an `async`, non-`@MainActor` func is _guaranteed_ to never run on the main thread, even by coincidence? I bet that is true currently and perhaps always will be, although I feel like semantically there is no particular guarantee there

Reply:
Yeah, that's fair. The assertion isn't really providing significant value anyway, so I'll just remove it.